core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -305,7 +305,7 @@ private[spark] class MapOutputTrackerMaster(conf: SparkConf)
 
     if (mapStatuses.contains(shuffleId)) {
       val statuses = mapStatuses(shuffleId)
-      if (statuses.nonEmpty) {
+      if (statuses.nonEmpty && statuses.exists(_ != null)) {
```

Comment:
nonEmpty is just to judge the array length != 0
but we will register array[partitionSize] for each shuffle when job submit.
So statuses,nonEmpty is always true